lib/matplotlib/cbook.py

```diff
@@ -658,6 +658,49 @@ def __setstate__(self, state):
         self.extend(state['seq'])
 
 
+class IgnoredKeywordWarning(UserWarning):
+    """
+    A class for issuing warnings about keyword arguments that will be ignored
+    by matplotlib
+    """
+    pass
+
+
+def kwarg_or_local(kwargs, kwarg_name, local_var=None):
+    """
+    Enforces the priority of a local variable over a potentially conflicting
+    argument from a kwargs dict
+
+    Parameters:
+    ------------
+        kwargs: dict
+            Dictionary of keyword arguments; may be modified in place
+
+        kwarg_name: str
+            Name of the keyword argument to process; kwargs[kwarg_name] will be
+            removed in place if it exists
+
+        local_var: any object, optional
+            Local variable
+
+    Returns:
+    ---------
+        out: any object
+            If local_var is None then kwargs[kwarg_name] will be returned,
+            otherwise local_var will be returned and an IgnoredKeywordWarning
```

Comment:
This is not correct.  -> "otherwise local_var will be returned. If kwargs[kwarg_name] has been overridden by local_var, an IgnoredKeywordWarning will be raised."  Actually, I think the sentence about the warning should be in a separate "Raises:" block, shouldn't it?